### PR TITLE
Reduce stats report size, remove all 0-value snapshots, use v to replace value

### PR DESCRIPTION
**Pull request by justinlin-linkedin**
There are a couple low hanging fruits to reduce the size of the stats report to helix
1) remove all the 0-value snapshots
2) use "v" to replace "value" so the key would be shorter.

Hopefully this would make report a bit smaller.

### CODE REVIEW

**Review by lightningrob — APPROVED** [review]
LGTM

File: ambry-api/src/test/java/com/github/ambry/server/StatsSnapshotTest.java [hunk 1]
```diff
@@ -66,5 +66,147 @@ public void deserializeStatsSnapshotTest() throws IOException {
...
+        }));
+        put("L1_K2", new StatsSnapshot(0L, new HashMap<String, StatsSnapshot>() {
+          {
+            put("L2_K3", new StatsSnapshot(70L, null));
+            put("L2_K4", new StatsSnapshot(0L, null));
+          }
+        }));
+        put("L1_K3", new StatsSnapshot(0L, new HashMap<String, StatsSnapshot>() {
+          {
+            put("L2_K5", new StatsSnapshot(20L, new HashMap<String, StatsSnapshot>() {
```
Comment (jsjtzyy) [inline ambry-api/src/test/java/com/github/ambry/server/StatsSnapshotTest.java:166]: In practice, `L2_K5` initial value `20L` shouldn't happen, right?

**Review by jsjtzyy — APPROVED** [review]
LGTM

File: ambry-api/src/test/java/com/github/ambry/server/StatsSnapshotTest.java (hunk 1 above)
Comment (justinlin-linkedin) [reply ambry-api/src/test/java/com/github/ambry/server/StatsSnapshotTest.java]: that's right, I was just being lazy to calculate the values for these snapshots so I call `updateValue` before `removeZeroValueSnapshots`. Anyway, I will put the right values in the constructor.

**Review by jsjtzyy — APPROVED** [review]

**Pull request merged**